src/rust/engine/fs/store/src/snapshot_ops.rs

```diff
@@ -425,17 +425,19 @@ impl IntermediateGlobbedFilesAndDirectories {
             if (*wildcard == *DOUBLE_STAR_GLOB) || (*wildcard == *SINGLE_STAR_GLOB) {
               // Here we short-circuit all cases which would swallow up a directory without
               // subsetting it or needing to perform any further recursive work.
+              let has_ignores = exclude.has_ignored_paths(&directory_path);
```

Comment:
This will be a relatively expensive method: maybe move calling it down to: `if short_circuit && !exclude.has_ignored_paths(&directory_path) { .. }`. Could then rename the `let short_circuit` flag to `let is_short_circuit_pattern`.